src/auth/auth-manager.js

```diff
@@ -0,0 +1,166 @@
+ */
+
+/**
+ * Given a valid email/password, generates and returns a JWT.
+ * @param {string} email - The user's email address.
+ * @param {string} password - The user's password.
+ * @param {loginResponse} next - The callback function to run after this function
+ *     finishes.
+ */
+const login = (email, password, next) => {
+  User.findOne({ email: email }, (err, user) => {
+    if (err || !user) {
+      next(err || { err: "Auth failed." });
+      return;
+    }
+
+    user.comparePassword(password, (err, isMatch) => {
+      if (err) {
+        next(err);
+        return;
+      }
+
+      Session.findOne({ email: email }, (err, session) => {
+        if (err || session) {
+          next(err, session.token);
+          return;
+        }
+
+        // Create a JWT.
+        const jwtData = {
+          email: email,
+          role: user.role,
+          isApproved: user.isApproved
+        };
+
+        // Sign and save the session.
+        jwt.sign(jwtData, process.env.SECRET, { algorithm: 'HS256' }, (err, token) => {
+          const session = new Session({
+            email: email,
+            token: token
+          });
+          session.save((err) => { next(err, token); });
+        });
+      });
+    });
+  });
+};
+
+/**
+ * Callback for sending the response to the client.
+ *
+ * @function logoutResponse
+ * @param {Object} err - The error.
+ * @param {string} err.err - The error message.
+ */
+
+/**
+ * Delete the user's JWT from the DB.
+ * @param {string} email - The user's email address.
+ * @param {logoutResponse} next - The callback function to run after this function
+ *     finishes.
+ */
+const logout = (email, next) => {
+  // Delete the session from the DB.
+  Session.findOne({ email: email }, (err, session) => {
+    if (err || !session) {
+      next(err);
+      return;
+    }
+    session.remove(next);
+  });
+};
+
+/**
+ * Callback for sending the response to the client.
+ *
+ * @function changePasswordResponse
+ */
+
+/**
+ * Delete the user's JWT from the DB.
+ * @param {string} email - The user's email address.
+ * @param {string} newPassword - The user's new password.
+ * @param {changePasswordResponse} next - The callback function to run after this function
+  *     finishes.
+ */
+const changePassword = (email, password, next) => {
+  // TODO(the-pat): Update the user's password and regen the JWT.
+  next();
+}
+
+/**
+ * Callback for sending the response to the client.
+ *
+ * @function approveResponse
+ * @param {(string|Object)} err - The error.
+ */
+
+/**
+ * Given an email, approve a User.
+ * @param {string} email - The user's email address.
+ * @param {approveResponse} next - The callback function to run after this function
+  *     finishes.
+ */
+const approve = (email, next) => {
+  AuthUser({email: email}, (err, user) => {
+    if (err) {
+      next(err);
+      return;
+    }
+
+    user.isApproved = true;
+    user.save((err) => { next(err); });
+  });
+};
```

Comment:
I think the intention was to eventually have a ```user/approve/:uid``` endpoint, and before then, approval would just be done through the ```modifyUser()``` method.

Since user approval is more or less outside the scope of auth functionality, I'd leave the ```approveUser()``` method/endpoint out of this PR entirely.